### DISCUSSION TITLE
3Phase Voltage drifting

### DISCUSSION THREAD

**Discussion by Xsez** [emporia-vue-local/esphome#197] [Q&A] [answered]
Greetings,

I have successfully flashed my device, but can't get the voltage calibrated since its drifting over a long time by a significant amount.

These are the voltage readings after a few hours:
![image](https://github.com/emporia-vue-local/esphome/assets/8041173/9519e25d-5ee5-4ad2-b947-c56fd484ad47)
![image](https://github.com/emporia-vue-local/esphome/assets/8041173/f5c74fd7-be69-4637-8082-33570e764a4d)
![image](https://github.com/emporia-vue-local/esphome/assets/8041173/99994757-a061-4bb5-9e2d-6604f6e31e30)

This, of course, causes the power readings to drift as well.

I had the device working for a day with original firmware, to try it out. Direct Voltage readings were not available in the emporia app, but the power readings were not drifting and made sense.

Initially, I needed to double the voltage calibration for phase 1 to .044 to read ~230V.
After letting the system sit for a few minutes i calculated these cal values: 0.0426, 0.0156, 0.0174. As you can see from the graphs these values were quite okay in the beginning but now drifted completely away. Measured Voltage with Fluke Multimeter is constantly at 235-237V.

Also the readings are quite noisy. If i remove the averaging filter, they jump around +-50V. You can see some noise in the graphs as well and that is with averaging filter. Is that normal?

Did I break something? Do i need to let the system settle for a longer time? Any help is apreaciated.

**Comment by flaviut** [comment, marked as answer]
Not at all normal. There's no settling time, it should just work instantly and definitely not drift.

This almost seems like there's some loose connection. Have you double-checked the connection to the panel, that all the breakers (IDK how y'all do it in Germany) are fully seated, etc?

**Reply by Xsez** [reply]
The neutral terminal, where i connected the white wire, was not connected to N at all. Wire was just floating. Interesting, that the vue booted up at all.
Now the signal seems perfectly stable and I'm back to 0.022 cal values for each phase.

I feel very stupid now... I'm terribly sorry to have bothered you...

**Reply by flaviut** [reply]
No worries! Someone else will certainly run into this again, and you've described the symptoms very well!
